composer/callbacks/speed_monitor.py

```diff
@@ -63,106 +63,106 @@ class SpeedMonitor(Callback):
-            "wall_clock/train": self.wall_clock_train,
-            "wall_clock/val": self.wall_clock_val,
-            "wall_clock/total": self.wall_clock_total,
-            "epoch_duration": current_time - self.epoch_start_time,
-            "batch_durations": [current_time - x for x in self.batch_end_times],
-            "batch_num_samples": self.batch_num_samples,
+            "epoch_start_num_samples": self.epoch_start_num_samples,
+            "epoch_start_wct": self.epoch_start_wct,
+            "batch_start_num_samples": self.batch_start_num_samples,
+            "batch_start_wct": self.batch_start_wct,
+            "rolling_batch_times_and_num_samples": self.rolling_batch_times_and_num_samples,
+            "running_wct": self.running_wct,
+            "running_num_samples": self.running_num_samples,
+            "total_eval_wct": self.total_eval_wct,
         }
 
     def load_state_dict(self, state: Dict[str, Any]) -> None:
-        self.loaded_state = state
+        self._loaded_state = state
 
     def _load_state(self) -> None:
-        current_time = time.time()
-        if self.loaded_state is not None:
-            self.train_examples_per_epoch = self.loaded_state["train_examples_per_epoch"]
-            self.wall_clock_train = self.loaded_state["wall_clock/train"]
-            self.wall_clock_val = self.loaded_state["wall_clock/val"]
-            self.wall_clock_total = self.loaded_state["wall_clock/total"]
-            self.epoch_start_time = current_time - self.loaded_state["epoch_duration"]
-            self.batch_end_times = deque([current_time - x for x in self.loaded_state["batch_durations"]],
-                                         maxlen=self.window_size + 1)
-            self.batch_num_samples = self.loaded_state["batch_num_samples"]
-            self.loaded_state = None
+        if self._loaded_state is not None:
+            self.epoch_start_num_samples = self._loaded_state["epoch_start_num_samples"]
+            self.epoch_start_wct = self._loaded_state["epoch_start_wct"]
+            self.batch_start_num_samples = self._loaded_state["batch_start_num_samples"]
+            self.batch_start_wct = self._loaded_state["batch_start_wct"]
+            self.rolling_batch_times_and_num_samples = deque(
+                [x for x in self._loaded_state["rolling_batch_times_and_num_samples"]],
+                maxlen=self.window_size,
+            )
+            self.running_wct = self._loaded_state["running_wct"]
+            self.running_num_samples = self._loaded_state["running_num_samples"]
+            self.total_eval_wct = self._loaded_state["total_eval_wct"]
+            self._loaded_state = None
 
-    def batch_start(self, state: State, logger: Logger) -> None:
+    def epoch_start(self, state: State, logger: Logger):
         del logger  # unused
         self._load_state()
-        self.batch_start_num_samples = state.timestamp.sample
+        self.rolling_batch_times_and_num_samples.clear()
```

Comment:
do we need to clear across the epoch boundary?

Reply:
Nope, don't need to -- this is how it was done before, but just removed it.